### DISCUSSION TITLE
how to decode topics as args

### DISCUSSION THREAD

**Discussion by tunnckoCore** [wevm/viem#852] [Question]
I want the decoded args of an event which is received through the watchEvent / watchContractEvent, like it happens with the results from getFilterLogs and getFilterChanges.

ideas? i can look at the code but my arm hurts and barely move it and barely write.

**Comment by tunnckoCore** [comment]
cc @jxom, ideas?

**Comment by tunnckoCore** [comment]
![Screenshot_20230707_213519_Brave](https://github.com/wagmi-dev/viem/assets/5038030/a5135d8a-6a4d-4e0b-8a3d-47dac6315c22)

Seems like it's happening with the `decodeEventLog` and `formatLog`, but they are not exposed to the public API?

**Comment by tunnckoCore** [comment]
Hm. Okay, they are actually exported from the src/index which is great. But they just are not in the docs i guess.

**Comment by tunnckoCore** [comment]
Oh okay, they are. I didn't seen them.

![image](https://github.com/wagmi-dev/viem/assets/5038030/75bb0374-969d-45df-ae62-0028499902f6)

**Reply by jxom** [reply]
The search bar is also handy 😃
